### PR TITLE
add availability date type component link

### PR DESCRIPTION
**Pull request by patel-kajal**
[US131930](https://rally1.rallydev.com/#/110294864140d/custom/110851118712?detail=%2Fuserstory%2F609654871663)

Mock up: https://xd.adobe.com/view/3ab11052-f6a1-4477-ab60-faecdcc72caa-6fe2/screen/1e275c14-9722-48aa-b9f7-cd9afd17d513/?fullscreen

PR1 by me and @adkulas 

This first PR will just render the "Before Start: <date type>" and "After End: <date type>" links to the activity availability date editor component when the availability date type property is present on the date entity.

recording

Follow up:
PR1 - add availability date component link
[PR2](https://github.com/BrightspaceHypermediaComponents/activities/pull/2004) - add new dialog component
[PR3](https://github.com/BrightspaceHypermediaComponents/activities/pull/2008) - some refactoring and clean up

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-availability-dates-editor.js [hunk 1]
```diff
@@ -32,39 +49,81 @@ class ActivityAvailabilityDatesEditor extends (ActivityEditorMixin(LocalizeActiv
...
+				</div>
+				<d2l-validation-custom
+					for="end-date-input"
+					@d2l-validation-custom-validate=${this._validateEndDate}
+					failure-text="${endDateError}">
+				</d2l-validation-custom>
+
+			`;
+		}else{
+			return html`
```
Comment (patel-kajal) [inline components/d2l-activity-editor/d2l-activity-availability-dates-editor.js:98]: this is the original html without the availability type

File: components/d2l-activity-editor/d2l-activity-availability-dates-editor.js [hunk 2]
```diff
@@ -32,39 +49,81 @@ class ActivityAvailabilityDatesEditor extends (ActivityEditorMixin(LocalizeActiv
...
-				@change="${this._onEndDatetimeChanged}"
-				?disabled="${!canEditDates}">
-			</d2l-input-date-time>
-			<d2l-validation-custom
-				for="end-date-input"
-				@d2l-validation-custom-validate=${this._validateEndDate}
-				failure-text="${endDateError}">
-			</d2l-validation-custom>
-		`;
+		if (startDateType || endDateType){
```
Comment (patel-kajal) [inline components/d2l-activity-editor/d2l-activity-availability-dates-editor.js:59]: the idea here is that if the HM api returns the startDateType/endDateType property, we can assume the UX flag is on and the availability types should be used

File: components/d2l-activity-editor/d2l-activity-availability-dates-editor.js [hunk 3]
```diff
@@ -84,7 +143,9 @@ class ActivityAvailabilityDatesEditor extends (ActivityEditorMixin(LocalizeActiv
 		const datesEntity = {
 			canEditDates: true,
 			startDate: null,
+			startDateType: null,
 			endDate: null,
+			endDateType: null,
```
Comment (patel-kajal) [inline components/d2l-activity-editor/d2l-activity-availability-dates-editor.js:148]: when the HM api work is done, I think the expectation is that the startDateType and endDateType properties will be returned and populated on the entity when the flag is on. This part of the code will need to be update when that work is done

File: components/d2l-activity-editor/d2l-activity-availability-dates-editor.js [hunk 4]
```diff
@@ -32,39 +49,81 @@ class ActivityAvailabilityDatesEditor extends (ActivityEditorMixin(LocalizeActiv
...
+					id="end-date-input"
+					label="${this.localize('editor.endDate')}"
+					time-default-value="endOfDay"
+					value="${endDate}"
+					@change="${this._onEndDatetimeChanged}"
+					?disabled="${!canEditDates}">
+				</d2l-input-date-time>
+				<div ?hidden=${!endDate} id="end-date-type-text">
+					<span class="d2l-body-small">${this.localize('editor.afterEndDate')}</span>
+					<d2l-link href="#" target="_blank" small>${this.localize('editor.lblVisibleWithAccessRestricted')}</d2l-link>
```
Comment (patel-kajal) [inline components/d2l-activity-editor/d2l-activity-availability-dates-editor.js:88]: the type here right now is hardcoded to access restricted, but should either be updated to whatever is returned from the HM api for the entity or set to the default value

Comment (pureooze) [reply components/d2l-activity-editor/d2l-activity-availability-dates-editor.js:81]: Also I'm not sure we need a separate if else statement for this, can we just do an eval on the `hidden` condition and return the value of the end date type from a method? Something like [this](https://github.com/BrightspaceHypermediaComponents/activities/blob/b21d19c943764073deffa15397a3dbeca549503e/components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-submission-and-completion.js#L389-L410)

**Review by pureooze — APPROVED** [review]

File: components/d2l-activity-editor/d2l-activity-availability-dates-editor.js (hunk 4 above)
Comment (adkulas) [reply components/d2l-activity-editor/d2l-activity-availability-dates-editor.js:81]: I like this idea better, we will clean this up in a follow PR

**Review by astadelbauer — APPROVED** [review]

**Pull request merged**